ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetriever.java

```diff
@@ -218,8 +243,15 @@ public synchronized void addSidecar(final DataColumnSidecar sidecar) {
         existingSidecarsByColIdx.put(sidecar.getIndex(), sidecar);
         if (existingSidecarsByColIdx.size() >= recoverColumnCount) {
           // TODO: Make it asynchronously as it's heavy CPU operation
-          recover();
-          recoveryComplete();
+          try (MetricsHistogram.Timer ignored =
```

Comment:
There are 3 types of DataColumnSidecars recovery in Teku currently:
1. Recovery during sync. When new node joins network or lost head for a lot slots and need to sync blocks and do DA for old blocks. It's here in `RecoveringSidecarRetriever`
2. Recovery from EL blobs during gossip, when the node is on head. It works only for current block. It's also called distributed blob building or smth like this I think. It's in `BlockBlobSidecarsTrackersPoolImpl`
3. Recovery from 64+ `DataColumnSidecars` on gossip for current block. It happens in `DataColumnSidecarRecoveringCustodyImpl` specially designed for this kind of recovery.

You will not get anything in 1 if you are in sync even if some columns were withholded. Looking at metrics name and type, I guess you need to measure it in (3)